library-best-practices/JLBP-9.md

```diff
@@ -1,20 +1,21 @@
 [JLBP-9] Support the minimum Java version of your consumers
 -----------------------------------------------------------
 
-If a library B requiring Java 1.7 depends on a library A requiring Java 1.7, and
-library A bumps its minimum version to Java 1.8, then library B cannot pick up
-the new version of library A until it also bumps its own minimum Java
-version. This can result in a situation where the ecosystem is stuck on older
-library versions, preventing the pickup of important things like security fixes,
-and it can also lead to diamond dependency conflicts where no version works for
-all consumers. Thus, libraries should make sure their actively-maintained
-consumers advance their minimum version of Java before they advance it
-themselves. (Waiting for every last consumer could block an upgrade permanently,
-due to cases like abandoned projects.)
+Imagine that library B depends on library A and both work with Java 8.
+If library A releases a new version that requires Java 11, library B cannot upgrade
+to the new version of library A unless it also starts requiring Java 11. 
+If library B is not able to do that—for instance because it must work in an environment
+such as App Engine that does not yet support Java 11—it must keep using the older
+version of library A, missing out on security and bug fixes in the newer version.
 
-As an alternative, if you need to support both consumers on the old version of
-Java and consumers wanting features provided by a newer version of Java, a
-long-term support (LTS) release can be created to continue supporting the old
-Java version, and a new major version of the library can be released with a
-higher minimum Java version. The Java package and maven ID should be kept the
-same in the new major version.
+Thus, libraries should not require a new minimum Java version until
+actively-maintained dependents already require that version of Java.
+It is rarely possible to wait for all dependents to upgrade.
+Some projects are abandoned or lightly maintained and may never upgrade.
+
+As an alternative, to support both dependents on the old version of
+Java and dependents wanting features provided by a newer version of Java, 
+you can continue development in two forks, version N for the older version of
+Java and version N+1 for the newer version of Java. Typically in this case,
+the major version is incremented to distinguish the two versions.
+The Java package and Maven coordinates should be the same in the new major version.
```

Comment:
For the background of the last sentence, you may want to add reference to JLBP-6 "Package and artifact renaming rules".